src/main/java/gift/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package gift.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<String> handleValidationExceptions(MethodArgumentNotValidException ex) {
+        String errorMessage = ex.getBindingResult()
+                .getFieldErrors()
+                .stream()
+                .map(error -> error.getDefaultMessage())
+                .findFirst()
+                .orElse("유효성 검사 실패");
+        return new ResponseEntity<>(errorMessage, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(InvalidProductNameException.class)
+    public ResponseEntity<String> handleInvalidProductNameException(InvalidProductNameException ex) {
+        return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
```

Comment:
ResponseEntity 클래스에 편의를 위해 정의된 정적 팩토리 메서드를 활용해볼 수도 있을 것 같아요. 

```suggestion
        return ResponseEntity.badRequest().body(ex.getMessage());
```